Update Table and AdvancedTable content to make documentation more concise

LilithJames-HDS opened the pull request:
### :pushpin: Summary

If merged, this PR will update the guidelines for the Table and Advanced Table documentation to make the content more streamlined and concise

### :link: External links
Jira ticket: [HDS-4420](https://hashicorp.atlassian.net/browse/HDS-4420)

***

### :eyes: Component checklist

- [ ] Percy was checked for any visual regression
- [ ] A changelog entry was added via [Changesets](https://github.com/changesets/changesets) if needed (see [templates here](https://hashicorp.atlassian.net/wiki/spaces/HDS/pages/3243114706/Changelog+authoring+best+practices#Templates))

:speech_balloon: Please consider using [conventional comments](https://conventionalcomments.org/) when reviewing this PR.

<details>

<summary>:clipboard: PCI review checklist</summary>

<!-- heimdall_github_prtemplate:grc-pci_dss-2024-01-05 -->
- [ ] If applicable, I've documented a plan to revert these changes if they require more than reverting the pull request.
- [ ] If applicable, I've worked with GRC to document the impact of any changes to security controls.
  Examples of changes to controls include access controls, encryption, logging, etc.
- [ ] If applicable, I've worked with GRC to ensure compliance due to a significant change to the in-scope PCI environment.
  Examples include changes to operating systems, ports, protocols, services, cryptography-related components, PII processing code, etc.

</details>

[HDS-4420]: https://hashicorp.atlassian.net/browse/HDS-4420?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

heatherlarsen commented on website/docs/components/table/advanced-table/partials/guidelines/guidelines.md:
```diff
@@ -236,17 +229,16 @@ Striping enhances readability by alternating row colors, making it easier to sca
 
 ![The cell with column placement end and row placement end has a border radius set on the bottom right corner.](/assets/components/table/advanced-table/table-row-placement.png)
 
 ## Cells
```
[suggestion] There's additional content under the Cells header in the Table docs that feels particularly helpful. Should we consider adding it to the Advanced Table docs as well?

> For the user to scan, sort, and filter the table easily, each cell should contain a single piece of data. Having more than one piece of data in a cell makes it harder for users to navigate the relationships between headers and cells.

heatherlarsen commented on website/docs/components/table/table/partials/guidelines/guidelines.md:
```diff
@@ -159,45 +152,44 @@ When using striping in a Table, the first tinted row will be the second row of t
 ## Cells
 
-Cells are the core organizational container within a Table. In order for the user to be able to sort and read the table easily, each cell in a table should contain one piece of data. Having more than one piece of information inside a cell also makes it harder for users to navigate the relationships between headers and content. 
+In order for the user to sort and read the table easily, each cell in a table should contain one piece of data. Having more than one piece of information inside a cell makes it harder for users to navigate the relationships between headers and content. 
```
```suggestion
For the user to scan, sort, and filter the table easily, each cell should contain a single piece of data. Having more than one piece of data in a cell makes it harder for users to navigate the relationships between headers and cells.
```

heatherlarsen commented on website/docs/components/table/table/partials/guidelines/guidelines.md:
```diff
@@ -159,45 +152,44 @@ When using striping in a Table, the first tinted row will be the second row of t
 ### Density
 
-- We recommend using `medium` cell density by default.
+- Use `medium` density by default for balanced readability and display.
+- Choose `short` density for text-heavy tables to fit more rows on a page but can make tables harder to read and scan, so use it thoughtfully.
 - If content is complex or a smaller data set (e.g., a Table of basic user data), `tall` cell density allows for more breathing room surrounding the content.
```
```suggestion
- By default, use the `medium` density for balance and readability.
- To fit more rows on a page, use the `short` density. Use this only for text-heavy tables, as it can make them harder to scan.
- For a smaller dataset, e.g., basic user data, consider using the `tall` density to provide the content with more breathing room.
```

heatherlarsen commented on website/docs/components/table/table/partials/guidelines/guidelines.md:
```diff
@@ -159,45 +152,44 @@ When using striping in a Table, the first tinted row will be the second row of t
 
 ## Horizontal scrolling
-Use horizontal scrolling when columns in a Table need to expand beyond the standard Table container. We recommend only implementing horizontal scrolling when data becomes difficult to read in a single view. Use the [Advanced Table](components/table/advanced-table), when keyboard navigation and sticky columns ease the reading experience for large data-sets.
+Use horizontal scrolling when columns need to expand beyond the standard container. We recommend only implementing horizontal scrolling when data becomes difficult to read in a single view. Use the [Advanced Table](components/table/advanced-table), when keyboard navigation and sticky columns ease the reading experience for large data-sets.
```
```suggestion
Use horizontal scrolling when the number of columns exceeds the viewport or container. Use the [Advanced Table](components/table/advanced-table) when keyboard navigation and sticky columns ease the reading experience for large datasets.
```

heatherlarsen commented on website/docs/components/table/table/partials/guidelines/guidelines.md:
```diff
@@ -159,45 +152,44 @@ When using striping in a Table, the first tinted row will be the second row of t
 ## Multi-select
 
-A multi-select table includes checkboxes enabling users to select multiple rows in a table to perform bulk operations. Checking or unchecking the checkbox in the table header either selects or deselects the checkboxes on each row in the table body. Individual checkboxes in the rows can also be selected or deselected.
+Multi-select allows users to select multiple rows to perform bulk actions, such as deleting or exporting data. 
```
```suggestion
Multi-select allows users to select multiple rows to perform bulk actions, such as deleting or exporting data. Selection states are maintained across pagination and filtering. 
```

heatherlarsen commented on website/docs/components/table/table/partials/guidelines/guidelines.md:
```diff
@@ -159,45 +152,44 @@ When using striping in a Table, the first tinted row will be the second row of t
 A multi-select consists of: 
 
-1. A select all in a table is used in the table's header row. This acts as the parent checkbox, allowing the selection or deselection of all child rows in a single table simultaneously.
+1. A select all in the table's header row to allow the selection or deselection of all child rows simultaneously.
```
[suggestion] Since these have been turned into full sentences, I would remove line 186 and the numbering from this and line 192.

```suggestion
A "Select all" checkbox is used in the header row to allow the simultaneous selection or deselection of all child rows.
```

heatherlarsen commented on website/docs/components/table/table/partials/guidelines/guidelines.md:
```diff
@@ -159,45 +152,44 @@ When using striping in a Table, the first tinted row will be the second row of t
 ![Example of multi-select in a table header](/assets/components/table/multi-select-header.png)
 
-2. Row level select is used in each table row allowing for the selection of an individual row.
+2. Row level select allowing for the selection of an individual row.
```
```suggestion
Individual checkboxes added to each row allow for the selection of that row.
```

heatherlarsen commented on website/docs/components/table/table/partials/guidelines/guidelines.md:
```diff
@@ -210,9 +202,8 @@ For more details around using a multi-select Table, recommended patterns, and in
 This example covers basic multi-select table interactions. We're working to include more complex patterns for handling actions when multiple rows are selected and how this impacts filtering and pagination within a data set. For questions or concerns, please reach out to [#team-design-systems](https://hashicorp.enterprise.slack.com/archives/C7KTUHNUS)
 !!!
```
I suggest we remove this banner.

heatherlarsen commented on website/docs/components/table/table/partials/guidelines/guidelines.md:
```diff
@@ -210,9 +202,8 @@ For more details around using a multi-select Table, recommended patterns, and in
-- When individual rows are selected, the parent checkbox in the Table header changes to display an indeterminate state. 
-- When no or some rows (but not all) are selected in a single Table, clicking the parent checkbox in the Table header will change to display as checked and all rows on that page will be selected.
-- When all rows are selected in a single Table, the parent checkbox in the header appears as checked. Clicking the parent checkbox will deselect all rows on that page.
+- When individual rows are selected, the parent checkbox in the header displays an indeterminate state. 
```
```suggestion
- When individual rows are selected, the "Select all" checkbox in the header displays an indeterminate state. 
```

heatherlarsen commented on website/docs/components/table/table/partials/guidelines/guidelines.md:
```diff
@@ -210,9 +202,8 @@ For more details around using a multi-select Table, recommended patterns, and in
-- When no or some rows (but not all) are selected in a single Table, clicking the parent checkbox in the Table header will change to display as checked and all rows on that page will be selected.
-- When all rows are selected in a single Table, the parent checkbox in the header appears as checked. Clicking the parent checkbox will deselect all rows on that page.
+- When individual rows are selected, the parent checkbox in the header displays an indeterminate state. 
+- When no or some rows (but not all) are selected, clicking the parent checkbox in the header will change to checked and all rows on that page will be selected.
```
```suggestion
- When no or some rows (but not all) are selected, clicking "Select all" in the header will check the checkbox, and all rows on that page will be selected.
```

heatherlarsen commented on website/docs/components/table/table/partials/guidelines/guidelines.md:
```diff
@@ -210,9 +202,8 @@ For more details around using a multi-select Table, recommended patterns, and in
 
-- When individual rows are selected, the parent checkbox in the Table header changes to display an indeterminate state. 
-- When no or some rows (but not all) are selected in a single Table, clicking the parent checkbox in the Table header will change to display as checked and all rows on that page will be selected.
-- When all rows are selected in a single Table, the parent checkbox in the header appears as checked. Clicking the parent checkbox will deselect all rows on that page.
```
This content should be included as it's part of the interaction pattern that isn't defined by the above bullet points. Though it could stand to be updated to something like:

> When all rows are selected, clicking "Select all" in the header will uncheck the checkbox, and all rows on that page will be deselected.

LilithJames-HDS commented:
> This is a great first pass! I dropped a bunch of suggestions about how we might further want to reduce the content.

@heatherlarsen  Thank you! I think I addressed all your notes. Let me know if you have any other concerns or edits

dchyun commented on website/docs/components/table/table/partials/guidelines/guidelines.md:
```diff
@@ -8,20 +8,19 @@
 
 - As a layout mechanism. Instead, use [Flex](/layouts/flex) or [Grid](/layouts/grid) layout helpers.
-- When data would best be displayed in a scrollable format, with more levels of hierarchy, or when keyboard navigation is needed to achieve usability. Instead, use the [Advanced Table] (components/table/advanced-table)
+- When data requires scrolling, more levels of hierarchy, or when keyboard navigation is needed to achieve usability. Instead, use the [Advanced Table](components/table/advanced-table).
```
```suggestion
- When data requires scrolling, more levels of hierarchy, or when keyboard navigation is needed to achieve usability. Instead, use the [Advanced Table](/components/table/advanced-table).
```
[Issue] Broken link

dchyun commented on website/docs/components/table/table/partials/guidelines/guidelines.md:
```diff
@@ -159,60 +148,54 @@ When using striping in a Table, the first tinted row will be the second row of t
 
 ## Horizontal scrolling
-Use horizontal scrolling when columns in a Table need to expand beyond the standard Table container. We recommend only implementing horizontal scrolling when data becomes difficult to read in a single view. Use the [Advanced Table](components/table/advanced-table), when keyboard navigation and sticky columns ease the reading experience for large data-sets.
+Use horizontal scrolling when the number of columns exceeds the viewport or container. Use the [Advanced Table](components/table/advanced-table) when keyboard navigation and sticky columns ease the reading experience for large datasets.
```
```suggestion
Use horizontal scrolling when the number of columns exceeds the viewport or container. Use the [Advanced Table](/components/table/advanced-table) when keyboard navigation and sticky columns ease the reading experience for large datasets.
```
[Issue] Broken link

dchyun reviewed (commented):
Looking good to me, just a couple minor changes.

heatherlarsen commented on website/docs/components/table/advanced-table/partials/guidelines/guidelines.md:
```diff
@@ -261,30 +263,25 @@ If `hasStickyFirstColumn` is set to true or false in the Ember component, a cont
-**Nested rows limitations**
+**Unsupported feature**
 
 Multi-select and sorting are not supported for nested rows at this time.
```
```suggestion
Multi-select is not supported for nested rows at this time.
```

heatherlarsen commented on website/docs/components/table/advanced-table/partials/guidelines/guidelines.md:
```diff
@@ -18,13 +18,13 @@
 
 ### Sorting
```
Remove "sorting" from the warning under the multi-select section and add a warning to the Sorting section.
 
```suggestion
### Sorting

!!! Warning

**Unsupported feature**

Sorting is not supported for nested rows at this time.
!!!
```

heatherlarsen reviewed (approved):
Left two minor comments. Looks great though!

shleewhite commented on website/docs/components/table/table/partials/guidelines/guidelines.md:
```diff
@@ -159,60 +148,54 @@ When using striping in a Table, the first tinted row will be the second row of t
-Row placement determines the visual styling based on where the row is placed relative to other rows within the Table. Only cells with a column placement that is either `start` or `end` utilize the row placement property; column position `middle` does not utilize this property.
+The `rowPlacement` property determines the border radius of a cell. It is only available on cells where the `colPlacement` property is set to `start` or `end`. 
 
 ![Table row placement example](/assets/components/table/table-row-placement.png)
```
```suggestion
![](/assets/components/table/table-row-placement.png)
```

shleewhite commented on website/docs/components/table/table/partials/guidelines/guidelines.md:
```diff
@@ -159,60 +148,54 @@ When using striping in a Table, the first tinted row will be the second row of t
+- To fit more rows on a page, use the `short` density. Use this only for text-heavy tables, as it can make them harder to scan.
+- For a smaller dataset, e.g., basic user data, consider using the `tall` density to provide the content with more breathing room.
 
 ![Table cell density](/assets/components/table/table-density.png)
```
```suggestion
![](/assets/components/table/table-density.png)
```

shleewhite reviewed (commented):
Mostly took the opportunity to remove unhelpful alt text, but I did find one issue. Love that we're making these docs clearer!

heatherlarsen commented on website/docs/components/table/table/partials/guidelines/guidelines.md:
```diff
@@ -50,68 +49,66 @@ Columns use as much room as they need to use to fit their content unless a speci
 !!!
 
-#### Start alignment
+#### Left alignment
```
@jorytindall The Figma component used to use start/end, didn't it? Do you remember when/why that would have changed?

LilithJames-HDS replied:
No, I don't but I just looked back and it has been that way since we published 2.0 so this isn't a recent change.

jorytindall replied:
Thinking back I _do_ remember some discussion about this, but I can't remember the context. It did used to use start/end, which was intentionally agnostic to account for internationalization, but the argument in code is left/center/right. I imagine that's why it was changed.
